util/translate_fight.py

```diff
@@ -28,6 +31,145 @@
+            if cast_match and not re.search(r'\(.+\|.+\)', cast_match.group(0)):
+                cast_match = cast_match.group(0).split(':')
+                    # Here and for begincast we convert the hex ability ID to decimal,
+                    # since that's what XIVAPI expects. We then convert back to string
+                    # to avoid later complications.
+                if not str(int(cast_match[2], base=16)) in translatables['action_ids']:
+                    translatables['action_ids'].append(str(int(cast_match[2], base=16)))
+                if not cast_match[1] in translatables['bnpcname']:
+                    translatables['bnpcname'].append(cast_match[1])
+                continue
+
+            begincast_match = e_tools.is_tl_line_begincast(sync_match.group(0).split('/')[1])
+            if begincast_match and not re.search(r'\(.+\|.+\)', begincast_match.group(0)):
+                if not begincast_match.group(1) in translatables['action_ids']:
+                    translatables['action_ids'].append(str(int(begincast_match.group(1), base=16)))
+                if not begincast_match.group(2) in translatables['bnpcname']:
+                    translatables['bnpcname'].append(begincast_match.group(2))
+                continue
+
+            # FIXME: The is_tl_line_buff() helper regex is currently bugged,
+            # and will NOT correctly return the buff/debuff.
+            # The second capture group will eat the source/caster name up with the buff.
+            # This line split  nonsense is to work around that
+            buff_match = e_tools.is_tl_line_buff(sync_match.group(0))
+            if buff_match:
+                buff_match = buff_match.group(2).split(' from')[0]
+                if not buff_match in ['status']:
+                    translatables['status'].append(buff_match)
+                continue
+
+            # Currently only looks for zone seal messages from log lines.
+            # FIXME: Add other log line possibilities here if necessary.
+            log_match = e_tools.is_tl_line_log(sync_match.group(1))
+            if log_match and 'will be sealed off' in log_match.group(2):
+                if not log_match.group(2).split(' will be')[0] in translatables['PlaceName']:
+                    translatables['PlaceName'].append(log_match.group(2).split(' will be')[0])
+
+            add_match = e_tools.is_tl_line_adds(sync_match.group(1))
+            if add_match:
+                if not add_match.group(1) in translatables['bnpcname']:
+                    translatables['bnpcname'].append(add_match.group(1))
+    return translatables
+
+def compile_and_send(translatables):
+    # XIVAPI/cafemaker allows for returning multiple items in one request if the item IDs are known.
+    IDstr = ",".join(translatables['action_ids'])
+
+    # These columns will be used for every XIVAPI request, so we alias them now.
+    columns = 'Name_en,Name_de,Name_fr,Name_ja'
```

Comment:
Can you build this from `xiv_lang_map` values?

Reply:
Good point, one less configuration update location.